src/sparsematrix.jl

```diff
@@ -1270,6 +1270,28 @@ function sparse!(I::AbstractVector{Ti}, J::AbstractVector{Ti},
             Vector{Ti}(undef, n+1), Vector{Ti}(), Vector{Tv}())
 end
 
+"""
+    SparseArrays.sparse!(I, J, V, [m, n, combine])
+
+Variant of `sparse!` that re-uses the input vectors `I`, `J`, `V` for the matrix storage.
```

Comment:
Maybe it is worth pointing out that some temporary arrays will still be allocated.